gufe/protocols/protocoldag.py

```diff
@@ -3,14 +3,93 @@
 
 import abc
 from typing import Iterable, List, Dict, Set, Optional, Union
+from os import PathLike
+import tempfile
 
 import networkx as nx
 
-from .protocolunit import ProtocolUnit
-from .results import ProtocolDAGResult, ProtocolDAGFailure
+from ..tokenization import GufeTokenizable
+from .protocolunit import ProtocolUnit, ProtocolUnitResult
 
 
-class ProtocolDAG:
+class ProtocolDAGResult(GufeTokenizable):
+    """Result for a single `ProtocolDAG` execution.
+
+    There may be many of these in a given `ResultStore` for a given `Transformation`.
+    Data elements from these objects are combined by `Protocol.gather` into a
+    `ProtocolResult`.
+
+    Attributes
+    ----------
+    name : str
+        Unique identifier for this `ProtocolDAGResult`.
+    graph : nx.DiGraph
+        The `ProtocolUnit`s, with dependencies set, as a networkx `DiGraph`.
+        Each `ProtocolUnit` features its `ProtocolUnitCompletion` as a `result` attribute.
+
+    """
+    name: Optional[str]
+    graph: nx.DiGraph
+
+    def __init__(self, *, name=None, graph):
+        self._name = name
+        self._graph = graph
+
+    def _defaults(self):
```

Comment:
Shouldn't this be the base class default?

Reply:
It's an abstractmethod in `GufeTokenizable`, so we have to override it for subclasses. I think this is appropriate, as defining defaults generally is hard and should be made on a per-subclass basis.